src/bin2pat/bin2pat.cpp

```diff
@@ -81,6 +83,37 @@ void processArgs(
 				return;
 			}
 		}
+		else if (args[i] == "-l" || args[i] == "--list") {
+			// Ensure -l --list is not the last thing in args
+			if (&args[i] == &args.back()) {
+				printErrorAndDie("input file missing");
+				return;
+			}
+
+			std::ifstream inputObjects(args[++i]);
+			std::string object;
+
+			if (inputObjects.is_open()) {
```

Comment:
I suggest moving the `if` and `else` clauses around to make the code more readable and reduce the indentation level. We could do this by converting
```cpp
if (inputObjects.is_open()) {
    // Read LIST_FILE until EOF
    while (inputObjects >> object) {
        // Ensure file exists before proceeding
        if(!FilesystemPath(object).isFile()) {
            printErrorAndDie("argument '" + args[i]
                + "' contains the filename '" + object
                + "' which is not a valid file");
            return;
        }
        else {
            inPaths.push_back(object);
        }
    }
}
else {
    printErrorAndDie("LIST_FILE '" + args[i]
        + "' is not a valid file");
    return;
}
```
into
```cpp
if (!inputObjects) {
    printErrorAndDie("LIST_FILE '" + args[i]
        + "' cannot be opened");
    return;
}

// Read LIST_FILE until EOF
while (inputObjects >> object) {
    // Ensure file exists before proceeding
    if(!FilesystemPath(object).isFile()) {
        printErrorAndDie("argument '" + args[i]
            + "' contains the filename '" + object
            + "' which is not a valid file");
        return;
    }
    else {
        inPaths.push_back(object);
    }
}
```
What do you think about this? This is just a suggestion, so if you like the original version, feel free to keep it.

Reply:
This, or something similar to it, will be implemented for readability when fixing the issue regarding inputObjects >> object.